src/vivarium_sodium_reduction/data/builder.py

```diff
@@ -128,3 +128,96 @@ def write_data_by_draw(artifact: Artifact, key: str, data: pd.DataFrame):
         data = data.reset_index(drop=True)
         for c in data.columns:
             store.put(f"{key.path}/{c}", data[c])
+
+
+def generate_consistent_rates(cause: str, art: Artifact, location: str, years: Optional[str]):
+    """Generates consistent rates for MOUD data.
+
+    Parameters
+    ----------
+    cause
+        the cause to make consistent rates for
+    art
+        The artifact to read from and write to.
+    location
+        The location associated with the data to load and the artifact to
+        write to.
+    years
+        The years to load data for.
+
+    """
+    # TODO: check if the consistent rates are already in the artifact, and if so, skip rest of this function
+
+    # copy metadata
+    for key in [
+        f"cause.{cause}.restrictions",
+        f"cause.{cause}.disability_weight",
+    ]:
+        data = art.load(key)
+        write_or_replace(art, key.replace(cause, f"{cause}_consistent"), data)
```

Comment:
I am a bit confused why the key would be existing? I would think that this would all happen during the process of writing the artifact (when you add the key to the artifact) and not going back later and looking at it again. What is the use case for this?

Reply:
It is for debugging, so that I can run this multiple times while iterating on the dismod code.